js/xt.js

```diff
@@ -0,0 +1,1072 @@
+                    'network': network,
+                    'fee': undefined,
+                    'deposit': deposit,
+                    'withdraw': withdraw,
+                    'limits': {
+                        'withdraw': {
+                            'min': undefined,
+                            'max': undefined,
+                        },
+                        'deposit': {
+                            'min': undefined,
+                            'max': undefined,
+                        },
+                    },
+                };
+            }
+            result[code] = {
+                'info': entry,
+                'code': code,
+                'id': currencyId,
+                'precision': undefined,
+                'name': name,
+                'active': true,
+                'deposit': depositEnabled,
+                'withdraw': withdrawEnabled,
+                'networks': networks,
+                'fee': undefined,
+                'limits': {
+                    'amount': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                },
+            };
+        }
+        return result;
+    }
+
+    safeNetwork (networkId) {
+        if (networkId === undefined) {
+            return undefined;
+        } else {
+            return networkId.toLowerCase ();
+        }
+    }
+
+    async fetchMarkets (params = {}) {
+        /**
+         * @method
+         * @name xt#fetchMarkets
+         * @description retrieves data on all markets for xt
+         * @param {object} params extra parameters specific to the exchange api endpoint
+         * @returns {[object]} an array of objects representing market data
+         */
+        // let type = undefined;
+        // [ type, params ] = this.handleMarketTypeAndParams ('fetchMarkets', undefined, params);
+        const query = this.omit (params, 'type');
+        // if (type === 'spot') {
+        //     // spot and swap ids are equal
+        //     // so they can't be loaded together
+        //     const spotMarkets = await this.fetchSpotMarkets (query);
+        //     return spotMarkets;
+        // }
+        return this.fetchSwapAndFutureMarkets (query);
+    }
+
+    async fetchSpotMarkets (params = {}) {
+        const response = await this.spotPublicGetDataApiV1GetMarketConfig (params);
+        //
+        //    {
+        //      "raca_xt":
+        //      {
+        //      "minAmount": 1,
+        //      "pricePoint": 7,
+        //      "coinPoint": 0,
+        //      "maker": 0.002,
+        //      "minMoney": 0,
+        //      "taker": 0.002
+        //      }
+        //    }
+        //
+        const result = [];
+        const ids = Object.keys (response);
+        for (let i = 0; i < ids.length; i++) {
+            const id = ids[i];
+            const market = this.safeValue (response, id);
+            const baseId = id.split ('_')[0];
```

Comment:
First we are doing two`splits` unnecessarily, do it once and store the result

Second, we should avoid direct access to arrays, in Python/PHP  this would crash the program if, for instance, the exchange introduced a new market that does not follow this structure (base_quote).

We should be doing this instead:
```Javascript
const baseId = this.safeString (ids, 0)
const quoteId = this.safeString (ids, 1)
```

Reply:
Like this?
```js
const splitId = id.split ('_')
const baseId = this.safeString (splitId, 0)
const quoteId = this.safeString (splitId, 1)
```

Reply:
@AlgoQ exactly